packages/components/src/InputText/InputText.tsx

```diff
@@ -52,12 +61,53 @@ function InputTextInternal(
 
   return (
     <FormField
+      {...props}
       type={props.multiline ? "textarea" : "text"}
       ref={inputRef}
-      {...props}
+      onChange={handleChange}
+      rows={rowRange.min}
     />
   );
 
+  function handleChange(newValue: string) {
+    props.onChange && props.onChange(newValue);
+
+    if (inputRef && inputRef.current instanceof HTMLTextAreaElement) {
+      resize(inputRef.current);
+    }
+  }
+
+  function getRowRange(): RowRange {
+    if (props.rows === undefined) {
+      return { min: 3, max: 3 };
+    } else if (typeof props.rows === "object") {
+      return { min: props.rows.min, max: props.rows.max };
+    } else {
+      return { min: props.rows, max: props.rows };
+    }
+  }
+
+  function resize(textArea: HTMLTextAreaElement) {
+    if (rowRange.min === rowRange.max) return;
```

Comment:
Return if no resizing is needed. This will handle the case where the values of rows is:

```
rows = {3}
```

or something like:

```
rows = {{min: 3, max: 3}}
```